### PR TITLE
coreutils: align formula with Linux

### PR DESCRIPTION
**Pull request by iMichka**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/coreutils.rb [hunk 1]
```diff
@@ -65,15 +71,24 @@ def install
       b2sum base32 chcon hostid md5sum nproc numfmt pinky ptx realpath runcon
       sha1sum sha224sum sha256sum sha384sum sha512sum shred shuf stdbuf tac timeout truncate
     ]
+    on_linux do
+      no_conflict << "dir"
+      no_conflict << "dircolors"
+      no_conflict << "vdir"
```
Comment (SMillerDev) [inline Formula/coreutils.rb:77]:
```suggestion
      no_conflict += [ "dir", "dircolors", "vdir"]
```

File: Formula/coreutils.rb [hunk 2]
```diff
@@ -41,10 +41,16 @@ class Coreutils < Formula
   def install
     system "./bootstrap" if build.head?
 
+    # Fix configure: error: you should not run configure as root
+    on_linux do
+      ENV["FORCE_UNSAFE_CONFIGURE"] = "1" if ENV["HOMEBREW_GITHUB_ACTIONS"]
+    end
```
Comment (SMillerDev) [inline Formula/coreutils.rb:47]: this build just really shouldn't run as root. But that is a problem for another time.

Comment (iMichka) [reply Formula/coreutils.rb:47]: Yes. If we find a clean way to handle running docker without root, and without breaking the whole CI, I'm game to remove all the root hacks we have. If anyone wants to start on working on a better solution, go ahead. I'll have time once the linux migration is done.

File: Formula/coreutils.rb [hunk 3]
```diff
@@ -65,15 +71,22 @@ def install
...
       bin.install_symlink "g#{cmd}" => cmd
       man1.install_symlink "g#{cmd}.1" => "#{cmd}.1"
     end
   end
 
   def caveats
+    msg = "Commands also provided by macOS"
+    on_linux do
+      "All commands"
+    end
```
Comment (carlocab) [inline Formula/coreutils.rb:87]: I'm not convinced this block does what (I think that) you think it does...

Comment (iMichka) [reply Formula/coreutils.rb]: That was actually a typo. Fixed. Good catch.

**Review by carlocab — APPROVED** [review]

**Pull request merged**